contracts/settling_game/library/library_calculator.cairo

```diff
@@ -43,4 +47,58 @@ namespace CALCULATOR:
         end
         return (happiness)
     end
+
+    func calculate_food{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        buildings : RealmBuildings, troop_population : felt
+    ) -> (food : felt):
+        alloc_locals
+
+        let House = BuildingsFood.House * buildings.House
+        let StoreHouse = BuildingsFood.StoreHouse * buildings.StoreHouse
+        let Granary = BuildingsFood.Granary * buildings.Granary
+        let Farm = BuildingsFood.Farm * buildings.Farm
+        let FishingVillage = BuildingsFood.FishingVillage * buildings.FishingVillage
+        let Barracks = BuildingsFood.Barracks * buildings.Barracks
+        let MageTower = BuildingsFood.MageTower * buildings.MageTower
+        let ArcherTower = BuildingsFood.ArcherTower * buildings.ArcherTower
+        let Castle = BuildingsFood.Castle * buildings.Castle
+
+        let food = House + StoreHouse + Granary + Farm + FishingVillage + Barracks + MageTower + ArcherTower + Castle
+
+        return (food - troop_population)
+    end
+
+    func calculate_population{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        buildings : RealmBuildings, troop_population : felt
+    ) -> (food : felt):
+        alloc_locals
+
+        let House = BuildingsPopulation.House * buildings.House
+        let StoreHouse = BuildingsPopulation.StoreHouse * buildings.StoreHouse
+        let Granary = BuildingsPopulation.Granary * buildings.Granary
+        let Farm = BuildingsPopulation.Farm * buildings.Farm
+        let FishingVillage = BuildingsPopulation.FishingVillage * buildings.FishingVillage
+        let Barracks = BuildingsPopulation.Barracks * buildings.Barracks
+        let MageTower = BuildingsPopulation.MageTower * buildings.MageTower
+        let ArcherTower = BuildingsPopulation.ArcherTower * buildings.ArcherTower
+        let Castle = BuildingsPopulation.Castle * buildings.Castle
+
+        let population = House + StoreHouse + Granary + Farm + FishingVillage + Barracks + MageTower + ArcherTower + Castle
+
+        return (population - troop_population)
+    end
+
+    # Returns coefficient for troop production in bp
+    func get_troop_coefficient{syscall_ptr : felt*, range_check_ptr}(
```

Comment:
so right now building a mage tower also aids archer troop production?

Reply:
correct. There is only 1 coefficient that impacts the overall squad.